### PR TITLE
Feature 7966

### PR DESCRIPTION
**Pull request by jlhicks4925**
### Redmine ticket(s)
https://devops.dchbx.org/redmine/issues/7966
https://devops.dchbx.org/redmine/issues/7932

### Local build result

bundle exec rake parallel:spec[4]
Finished in 11 minutes 23 seconds (files took 9.29 seconds to load)
4174 examples, 0 failures, 78 pending

bundle exec cucumber
58 scenarios (58 passed)
1001 steps (1001 passed)
7m17.926s

### Latest rebase/merge tag
tag: 1.8.37

---

### Data ticket(s) Followup
* (redmine links here - optional)

### Related Pull Requests
* (github links here - optional)

---

#### Peer Review (August 1st, 2016)
1. No questions are displayed when a non-allowable date is entered. 
2. rspec and cucumber tests are passing.
3. The branch name is feature-7966

#### Functional Testing
* (For testing locally)

#### Deployment
* (for release manager and/or build manager)

### CODE REVIEW

File: app/assets/javascripts/qle.js.erb [hunk 1]
```diff
@@ -28,7 +64,6 @@ $(function () {
     var splitedate = cdate.split('/');
     var cur_date = new Date(splitedate[2], splitedate[0]-1, splitedate[1]);
```
Comment (davidplappert) [inline app/assets/javascripts/qle.js.erb]: Should this date be pulled from Ruby w/ timekeeper?

File: app/controllers/insured/families_controller.rb [hunk 2]
```diff
@@ -336,4 +338,13 @@ def notice_upload_secure_message(notice)
     @person.inbox.messages << Message.new(subject: subject, body: body, from: 'DC Health Link')
     @person.save!
   end
+
+  def calculate_dates
```
Comment (vineelsonnathi) [inline app/controllers/insured/families_controller.rb]: @jlhicks4925 Please write a controller spec for this method.

File: app/controllers/insured/families_controller.rb [hunk 3]
```diff
@@ -145,6 +146,14 @@ def check_qle_date
...
   end
 
+  def check_move_reason
+    calculate_dates
+  end
+
+  def check_insurance_reason
+    calculate_dates
+  end
+
```
Comment (Kevin-Wei) [inline app/controllers/insured/families_controller.rb]: why the duplication here? @jlhicks4925

File: app/assets/javascripts/qle.js.erb [hunk 4]
```diff
@@ -1,3 +1,40 @@
...
+    });
+  }
+
+  function qle_question(url){
+      $.ajax({
+        type: "GET",
+        data:{date_val: $("#qle_date").val(), qle_id: $("#qle_id").val()},
+        url: "/insured/families/" + url
+      });
+  }
```
Comment (Kevin-Wei) [inline app/assets/javascripts/qle.js.erb]: lines 9-13 is overindented @jlhicks4925

File: app/views/insured/families/_qle_detail.html.erb [hunk 5]
```diff
@@ -68,3 +67,8 @@
     </div>
   </div>
 </div>
+<script>
+$(document).on('ready page:load', function () {
+  QLE.initialize();
+});
+</script>
```
Comment (Kevin-Wei) [inline app/views/insured/families/_qle_detail.html.erb]: This ok @withintheruins14 ?

File: app/assets/javascripts/qle.js.erb (hunk 4 above)
Comment (jlhicks4925) [inline app/assets/javascripts/qle.js.erb]: resolved @Kevin-Wei

File: app/controllers/insured/families_controller.rb (hunk 3 above)
Comment (jlhicks4925) [inline app/controllers/insured/families_controller.rb]: there's no duplication. check_move_reason and check_insurance_reason utilizes calculate_dates method

File: app/assets/javascripts/qle.js.erb (hunk 1 above)
Comment (jlhicks4925) [inline app/assets/javascripts/qle.js.erb]: it is utilizing Timekeeper

File: app/controllers/insured/families_controller.rb (hunk 2 above)
Comment (jlhicks4925) [inline app/controllers/insured/families_controller.rb]: specs are written for controller @vineelsonnathi

File: app/views/insured/families/_qle_detail.html.erb (hunk 5 above)
Comment (withintheruins14) [inline app/views/insured/families/_qle_detail.html.erb]: perfect

**Pull request merged**